server/channels/api4/outgoing_oauth_connection.go

```diff
@@ -328,3 +330,45 @@ func deleteOutgoingOAuthConnection(c *Context, w http.ResponseWriter, r *http.Re
 
 	ReturnStatusOK(w)
 }
+
+// validateOutgoingOAuthConnectionCredentials validates the credentials of an outgoing oauth connection by requesting a token
+// with the provided connection configuration. If the credentials are valid, the request will return a 200 status code and
+// if the credentials are invalid, the request will return a 400 status code.
+func validateOutgoingOAuthConnectionCredentials(c *Context, w http.ResponseWriter, r *http.Request) {
+	auditRec := c.MakeAuditRecord("validateOutgoingOAuthConnectionCredentials", audit.Fail)
+	defer c.LogAuditRec(auditRec)
+	c.LogAudit("attempt")
+
+	if !checkOutgoingOAuthConnectionWritePermissions(c) {
+		return
+	}
+
+	service, ok := ensureOutgoingOAuthConnectionInterface(c, whereOutgoingOAuthConnection)
```

Comment:
Slightly unrelated but it feels a bit weird to be passing this `whereOutgoingOAuthConnection`. Maybe it's something that could be part of the context.

Reply:
I was just following how it was done somewhere else, tbh. What would be the appropriate way of setting this out in the context?

Reply:
I think we could simply add a new `string` field to the context struct to track this information. Using the existing `ctx.Err` could be cleaner but may get tricky given it's a pointer. Anyhow, nothing blocking, just a thought.

Reply:
Oh, I thought there was already a way of doing it atm. I will leave it as it is for now and we can change that in a specific PR addressing just that.